### PR TITLE
Check for fractional offsets (closes #450, closes #439, closes #365)

### LINKED ISSUE #365: MoveAutomation is endless if end point coordinates are fractional
Sometimes [_getTargetClientPoint](https://github.com/DevExpress/testcafe/blob/master/src/client/runner/automation/playback/move.js#L349) returns fractional coordinates in Firefox. In this case a condition in [_isMovingFinished](https://github.com/DevExpress/testcafe/blob/master/src/client/runner/automation/playback/move.js#L275) never becomes true.

### LINKED ISSUE #439: Add 'not fractional' validation for 'number' options

### LINKED ISSUE #450: Incorrect error message when validating number for NaN
When validating a property that must be a number (e.g. `offsetX`) for `NaN` value, the error message will be:
`Action option offsetX is expected to be a number, but it was number.` 
instead of expected:
`Action option offsetX is expected to be a number, but it was NaN.`

It's because of `number()` validator ([here](https://github.com/DevExpress/testcafe/blob/ff3a7e506d8d91201054f394929d8e1c440ac9f5/src/test-run/commands/options.js#L15)):
```js
function number (option, val, positive) {
    var valType   = typeof val;
    var ErrorCtor = positive ? ActionPositiveNumberOptionError : ActionNumberOptionError;

    if (valType !== 'number')
        throw new ErrorCtor(option, valType);

    if (isNaN(val))
        throw new ErrorCtor(option, valType);
}
```
 `typeof NaN` is `"number"`, not `"NaN"`.

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @AlexanderMoskovkin, @helen-dikareva, @inikulin

### CODE REVIEW

**Comment by AndreyBelym** [comment]
@VasilyStrelyaev, please, check test names  [here](https://github.com/DevExpress/testcafe/pull/452/files#diff-9970ef94313c0fff2cc6328e4a279ad0R367) and [here](https://github.com/DevExpress/testcafe/pull/452/files#diff-9970ef94313c0fff2cc6328e4a279ad0R380), and also a comment [here](https://github.com/DevExpress/testcafe/pull/452/files#diff-27aea5773c389e89e84c3a8e9953640dR588)

File: test/client/fixtures/runner/api/drag-test.js [hunk 1]
```diff
@@ -364,7 +364,7 @@ $(document).ready(function () {
         );
     });
 
-    asyncTest('not int x or y argument raise error', function () {
+    asyncTest('non-numeric x or y argument raise error', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/api/drag-test.js]:
```
    asyncTest('non-numeric x or y argument raises an error', function () {
```

**Comment by VasilyStrelyaev** [comment]
\r-

File: src/errors/test-run/templates.js [hunk 2]
```diff
@@ -17,14 +17,14 @@ function markup (err, msgMarkup) {
...
+    [TYPE.actionIntegerNumberOptionError]: err => markup(err, `
+        Action option <code>${err.optionName}</code> is expected to be an integer number, but it was <code>${err.actualValue}</code>.
 
         ${err.getCallsiteMarkup()}
     `),
 
-    [TYPE.actionPositiveNumberOptionError]: err => markup(err, `
-        Action option <code>${err.optionName}</code> is expected to be a positive number, but it was <code>${err.actualValue}</code>.
+    [TYPE.actionPositiveIntegerNumberOptionError]: err => markup(err, `
+        Action option <code>${err.optionName}</code> is expected to be a positive integer number, but it was <code>${err.actualValue}</code>.
```
Comment (inikulin) [inline src/errors/test-run/templates.js]: `integer number`. `integer` already means `number`.

File: src/errors/test-run/type.js [hunk 3]
```diff
@@ -4,10 +4,10 @@
 // -------------------------------------------------------------
 
 export default {
-    actionNumberOptionError:         'actionNumberOptionError',
-    actionPositiveNumberOptionError: 'actionPositiveNumberOptionError',
-    actionBooleanOptionError:        'actionBooleanOptionError',
-    actionOptionsTypeError:          'actionOptionsTypeError',
+    actionIntegerNumberOptionError:         'actionIntegerNumberOptionError',
```
Comment (inikulin) [inline src/errors/test-run/type.js]: `IntegerNumber`

File: src/test-run/commands/options.js [hunk 4]
```diff
@@ -6,35 +6,35 @@
...
     var valType   = typeof val;
-    var ErrorCtor = positive ? ActionPositiveNumberOptionError : ActionNumberOptionError;
+    var ErrorCtor = positive ? ActionPositiveIntegerNumberOptionError : ActionIntegerNumberOptionError;
 
     if (valType !== 'number')
         throw new ErrorCtor(option, valType);
 
-    if (isNaN(val))
-        throw new ErrorCtor(option, valType);
+    if (isNaN(val) || !Number.isInteger(val))
```
Comment (inikulin) [inline src/test-run/commands/options.js]: `Number.isInteger` not available in IE and Safari. It's ES2017 proposal. Use this polyfill instead:
```js
var isInteger =  typeof value === 'number'
                        && isFinite(value)
                        && Math.floor(value) === value;
```

**Comment by inikulin** [comment]
\r-

File: src/test-run/commands/options.js (hunk 4 above)
Comment (AndreyBelym) [inline src/test-run/commands/options.js]: babel-transform-runtime polyfills that, below is a part of generated code of `options.js`:
```js
//...
var _isInteger = require('babel-runtime/core-js/number/is-integer');

var _isInteger2 = _interopRequireDefault(_isInteger);
//...

function integerNumber(option, val, positive) {
    var valType = typeof val === 'undefined' ? 'undefined' : (0, _typeof3.default)(val);
    var ErrorCtor = positive ? _testRun.ActionPositiveIntegerNumberOptionError : _testRun.ActionIntegerNumberOptionError;

    if (valType !== 'number') throw new ErrorCtor(option, valType);

    if (isNaN(val) || !(0, _isInteger2.default)(val)) throw new ErrorCtor(option, val);
}

```

Comment (inikulin) [inline src/test-run/commands/options.js]: Runtime is disabled for client code

Comment (AndreyBelym) [inline src/test-run/commands/options.js]: OK, didn't noticed that. So it means also no generators/`awaits` on a client, very sad. 
I'll use that check, it follows ECMA's [isInteger](https://tc39.github.io/ecma262/#sec-number.isinteger) definition.

Comment (inikulin) [inline src/test-run/commands/options.js]: > no generators/awaits on a client, very sad. 

Nah, Promises just fine =)

**Comment by AndreyBelym** [comment]
@testcafe-build-bot \retest

**Comment by AndreyBelym** [comment]
@testcafe-build-bot \retest

**Comment by AndreyBelym** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

File: src/client/runner/api/actions.js [hunk 5]
```diff
@@ -584,6 +584,11 @@ export function drag (what) {
                 options = options || {};
 
                 var { offsetX, offsetY } = getOffsetOptions(element, options.offsetX, options.offsetY);
+
+                //Need to round offsets due to GH-365
```
Comment (helen-dikareva) [inline src/client/runner/api/actions.js]: // NOTE: Need to round offsets due to GH-365

**Comment by helen-dikareva** [comment]
\r-, just one remark for consistency of comments

**Comment by AndreyBelym** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm

**Pull request merged**